src/hotspot/share/nmt/nmtNativeCallStackStorage.hpp

```diff
@@ -68,36 +69,46 @@ class NativeCallStackStorage : public CHeapObj<mtNMT> {
   };
 
 private:
-  struct Link : public ArenaObj {
-    Link* next;
+  struct Link;
+  using Allocator = ALLOCATOR<Link, mtNMT>;
+  using LinkPtr = typename Allocator::I;
+  LinkPtr nil() { return Allocator::nil; }
+
+  Allocator _allocator;
+
+  struct Link {
+    LinkPtr next;
     StackIndex stack;
-    Link(Link* next, StackIndex v)
+    Link(LinkPtr next, StackIndex v)
```

Comment:
Pre-existing. We should never modify the hash table, so the stackindex can be const.